RadixWallet/Features/DappInteractionFeature/Coordinator/DappInteractionFlow.swift

```diff
@@ -1156,3 +1156,14 @@ struct SavedPersonaDataInPersonaDoesNotMatchWalletInteractionResponseItem: Swift
 struct PersonaDataEntryNotFoundInResponse: Swift.Error {
 	let kind: PersonaData.Entry.Kind
 }
+
+private extension DappToWalletInteractionSubintentExpiration {
+	var timestamp: Date {
+		switch self {
+		case let .afterDelay(afterDelay):
+			Date().addingTimeInterval(TimeInterval(afterDelay.expireAfterSeconds))
```

Comment:
`Date()`? Shouldn't the timestamp be UTC?

Reply:
oh dearful dates and timezones.. Let me write down my dummy/cautious reasoning to validate the current logic makes sense.

In this case, the subintent will expire X seconds after signature. Let's use `600` (10 minutes) for this example, which we will set using the [test dApp](https://d1vuj67scmz8qj.cloudfront.net/).

It is currently `13:54 PM GMT+1` in Madrid. This means that when I get to this state, the `timestamp: Date` will return `14:04 GMT+1`. 

Given this `timestamp` is used on the `PollPreAuthorizationStatus+View`, we can validate that after signing the preAuth, the view says that it expires in `9 minutes` (it shows 10 minutes only for a second) and not in `-50 minutes` (as it would be in `GMT` timezone)

<img src=https://github.com/user-attachments/assets/4dfc23bc-f095-4136-b885-3e33a717bf87 width=150>



So far so good, now we need to make sure that the value sent to Sargon and then to the dApp makes sense. The process here is: 
- convert `Swift.Date` into `Sargon.Timestamp` (happens thanks to magic code in `uniffi.toml`)
- get amount of second since this timestamp and send it to dApp (done [here](https://github.com/radixdlt/sargon/pull/279/files#diff-090b4502eb51dbc6bffb6645ec282687461147f1e0fc282cfb1e1d4ddb5a3d71R33-R38))

Now, if we check the final result sent to the dApp we can see it is `1732799073`.
<img width="400" alt="Screenshot 2024-11-28 at 13 58 36" src="https://github.com/user-attachments/assets/9d80465f-c100-4704-940c-d4cb139efd5f">


Then, using a Playground I can confirm this value corresponds to `14:04` in my timezone:
<img width="833" alt="Screenshot 2024-11-28 at 14 00 09" src="https://github.com/user-attachments/assets/97ab195e-d1f0-440d-8610-0f74699013cb">

Or if I use a helper tool such as https://www.unixtimestamp.com/ we can verify it is `14:04 GMT+1` and `13:04 GMT`:
<img src=https://github.com/user-attachments/assets/6189156f-31bb-4d05-94c8-c210cd792b57 width=350>

Reply:
given that the same delay applies, regardless of the base, I think this could be safe, unless user goes and changes the date in the settings while the countdown is ticking 😄.